components/script/dom/window.rs

```diff
@@ -369,10 +370,12 @@ pub(crate) struct Window {
     /// Unminify Css.
     unminify_css: bool,
 
-    /// Where to load userscripts from, if any. An empty string will load from
-    /// the resources/user-agent-js directory, and if the option isn't passed userscripts
-    /// won't be loaded.
-    userscripts_path: Option<String>,
+    /// Where to load userscripts from, if any.
+    /// and if the option isn't passed userscripts won't be loaded.
+    userscripts_directory: Option<String>,
+
+    /// Inlined userscripts to run
+    userscripts: Vec<String>,
```

Comment:
Please don't add any more fields to `Window` for options and settings. These aren't necessary any longer now that script depends on `servo_pref`. Instead have the code in `userscripts.rs` read the options directly.